### PR TITLE
Update trim-self-contained.md

### PR DESCRIPTION
**Pull request by samsp-msft**
## Summary

Broke up big paragraph and provided a bit more detail.
Corrected trim argument.

### CODE REVIEW

**Comment by samsp-msft** [comment]
@jamshedd

**Review by LakshanF — COMMENTED** [review]
I believe the TrimMode is link for BlazorApps (what was there before)

File: docs/core/deploying/trim-self-contained.md [hunk 1]
```diff
@@ -9,9 +9,13 @@ ms.date: 04/03/2020
...
+
+The build time analysis engine provides warnings to the developer of code patterns that are problemmatic to detect which other code is required. Code can be annotated with attributes to tell the trimmer what else to include. Many reflection patterns can be replaced with build-time code generation using [Source Generators](https://github.com/dotnet/roslyn/blob/master/docs/features/source-generators.md).
+
+The trim mode for the applications can be configured via the TrimMode and will default (`copyused`) to bundle assemblies that are used in the application. Blazor WebAssembly applications will use a more aggressive mode (`trim`) that will trim unused code within assemblies. Trim analysis warnings give information on code patterns where a full dependency analysis was not possible. These warnings are suppressed by default and can be turned on by setting the flag, `SuppressTrimAnalysisWarnings`, to false. More information on the trim options available can be found at the [ILLinker page](https://github.com/mono/linker/blob/master/docs/illink-options.md).
```
Comment (adegeo) [inline docs/core/deploying/trim-self-contained.md:18]: Minor wording adjustments. We generally don't put a value in ``(`value`)`` syntax and instead call it out directly.

```suggestion
The trim mode for the applications is configured with the `TrimMode` setting. The default value is `copyused` and bundles referenced assemblies with the application. The `trim` value is used with Blazor WebAssembly applications and trims unused code within assemblies. Trim analysis warnings give information on code patterns where a full dependency analysis was not possible. These warnings are suppressed by default and can be turned on by setting the flag `SuppressTrimAnalysisWarnings` to `false`. For more information about the trim options available, see the [ILLinker page](https://github.com/mono/linker/blob/master/docs/illink-options.md).
```

**Review by adegeo — CHANGES_REQUESTED** [review]
Some minor things to fix.

**Review by adegeo — APPROVED** [review]
LGTM thanks!

**Pull request merged**